Release into Master

Joepieler opened the pull request:
@gerritvanos

crisvdn committed:
Missing encoders in constructor of beetle

crisvdn committed:
Integrated existing code from feature-rhino-turn with most up-to-date release.

crisvdn committed:
Added encode values in constructor of beetle.

crisvdn committed:
added test main loop

crisvdn committed:
Inverted speed of 1 motor so the rhino drives forward/backward properly.

crisvdn committed:
Turn function nowuses encoders to turn.

Joepieler committed:
Trying to make pid working but the arduino can't handel treads at this point in time

Joepieler committed:
resolderd the motor wires so now we they are same forward. also in code

crisvdn committed:
Implemented turn function with encoders

The rhino class is now compatible with the new updated UML class diagram. The rhino now uses the encoders to turn in a circle.

crisvdn committed:
Edited main test loop

Test loop for rhino turn is inside this main. Added documentation for each line what is done.

crisvdn committed:
Merge branch 'release' into feature-rhino-turn-with-encoder

crisvdn committed:
Refactored rhino class to be compatible after merge

Small changes were done to interface class due to a change in class diagram. After this merge, functions were edited / removed which were removed in the rhino class in this commit.

crisvdn committed:
Merge branch 'master' into feature-rhino-turn-with-encoder

Joepieler committed:
Work in progress tying to make te robot move in a stright line.

Joepieler committed:
Test move 1 meter in a straight line

Joepieler committed:
I have improved the code and made it a little more efficient. have an explanation for the different variable.

Joepieler committed:
test drive 50 cm in a straight line

Joepieler committed:
fixed main to canbus main

crisvdn committed:
Changed function variables to private varables

min_degrees and max_degrees are now constexpr and uint8_t to maximize the amount of memory usage

crisvdn committed:
adc_voltage is now constexpr

Changes made that were requested by PR

crisvdn committed:
Removed meaningless comments

crisvdn committed:
removed test code in main

The code is now ready and only uses the process function. Unused boolean also removed

crisvdn committed:
Merge branch 'feature-rhino-turn-with-encoder' of https://github.com/R2D2-2019/moving_platform into feature-rhino-turn-with-encoder

crisvdn committed:
Refactored calculation into variable

Calculation was done multiple times has been replaced by encoder_target_value. This variable holds the amount of revolutions the encoder needs to reach to make sure the turn of the rhino is complete

crisvdn committed:
Constexpr values are now static

A compile error was generated, this commit fixes the compile error about constexpr static variables.

crisvdn committed:
Removed unnecessary condition in while loop

crisvdn committed:
Added documentation inside turn function

Added documentation on the booleans inside the turn function.

crisvdn committed:
Changed while condition

Changed while condition so the code is more efficient and has one if statement less to check.

crisvdn committed:
Merge pull request #22 from R2D2-2019/feature-rhino-turn-with-encoder

Feature rhino turn with encoder

Joepieler committed:
feedback of Gerrit prossed made varaible private and fixed bug that the robot will move forever

Joepieler committed:
had merge conflict fixed it now

Joepieler committed:
edit discription of move function

Joepieler committed:
changed the total counters so they are not going out of scope

Joepieler committed:
fixed typo

Joepieler committed:
changed int to uint_fast64_t for better performance

Joepieler committed:
fixed typo

Joepieler committed:
maked kp constexpr

Joepieler committed:
fixed typo

Joepieler committed:
updated travis file approved by gerrit (lead)

Joepieler committed:
Merge branch 'release' into feature-move_certain_distance_beetle

Joepieler committed:
fixed compile errors seeing merge

Joepieler committed:
changed kp name to proportional_gain so it isn't confusting with kp from pid

Joepieler committed:
Merge pull request #25 from R2D2-2019/feature-move_certain_distance_beetle

Feature move certain distance beetle

Joepieler committed:
fixed turn side beetle

The beetle was turning the wrong side. -90 was right and 90 was left.

Joepieler committed:
Merge pull request #27 from R2D2-2019/hotfix-turn_beetle_wrong_way

fixed turn side beetle

JasperOosterbroek commented on code/headers/rhino.hpp:
```diff
@@ -0,0 +1,56 @@
+        void turn(int8_t degrees) override;
+        /**
+         * @brief
+         * This function will move the platform a certain distance
```
Distance has not been specified is this in meters, centimeters or even milimeters

Joepieler replied:
e91c3adcaa62a1dea3f09c75603d158f7fe5b4bd

JasperOosterbroek commented on code/src/beetle.cpp:
```diff
@@ -31,81 +31,163 @@ namespace r2d2::moving_platform {
+            if (counter_m1_total == encoder_rotations) {
+                qik_2s12v10_motorcontroller.brake_m1(0);
+            } else if (counter_m1_total < encoder_rotations) {
+                // the lowest the motor can power de motor can revieve is 10
```
Typos in this sentence

Joepieler replied:
c73c969c7b28468a09c5aac81b6f276bdcc33c5a

JasperOosterbroek commented on code/src/beetle.cpp:
```diff
@@ -31,81 +31,163 @@ namespace r2d2::moving_platform {
+
+            // Master slave method
+            // update every interval
+            if (hwlib::now_us() - tick > interval) {
```
Tick might not be the best variable name for this it is as far as I know an old time or previous time rather than a tick.

Joepieler replied:
previous_measurement_time kudo's to @JasperOosterbroek

Joepieler replied:
6db22dfa9e33e1c8a4d681eaa52cad9f30d3d5e2

JasperOosterbroek commented on code/headers/beetle.hpp:
```diff
@@ -43,5 +51,12 @@ namespace r2d2::moving_platform {
+         * @brief
+         * This function will move the platform a certain distance in cm
+         * @param    uint16_t: distance
+         * Minimal moving distance is 10 cm.
```
Is this meant in the param or in the brief description?

Joepieler replied:
param description

JasperOosterbroek commented on code/headers/rhino.hpp:
```diff
@@ -0,0 +1,56 @@
+         * Constructor rhino_c
+         * @param    qik_2s12v10_motorcontroller
+         * @param    comm
+         * @param    encode_m0
```
A lot of params in this class have not been specified, please add some explanation to the variables. The beetle.hpp should give a good example for what they should look like

Joepieler replied:
0d7eec6ad3a37e4736af6680c0bcd19e4c1a6d4b

crisvdn replied:
https://github.com/R2D2-2019/moving_platform/pull/29/commits/2a61229eecbf8d3bba04bcb6c30a3b43a947b3a9

JasperOosterbroek replied:
> [0d7eec6](https://github.com/R2D2-2019/moving_platform/commit/0d7eec6ad3a37e4736af6680c0bcd19e4c1a6d4b)

Why was this one also linked here?

gerritvanos replied:
error, misunderstanding of the comment

JasperOosterbroek replied:
That can always happen, the changes made in that hash are welcome since they were things I had originally missed.

JasperOosterbroek reviewed (changes requested):
Some minor spelling errors and a tricky variable name no major problems.

Joepieler committed:
updated brief function move rhino

Joepieler committed:
have sense improved

Joepieler committed:
changed variable tick to previous_measurement_time

Joepieler committed:
updated discription functions rhino

Joepieler committed:
changed encoders also in cpp

crisvdn committed:
Added doxygen

Added doxygen parameters explanation

crisvdn committed:
Merge pull request #29 from R2D2-2019/hotfix-pr_26

Hotfix pr 26

gerritvanos commented:
@JasperOosterbroek all feedback should be resolved by now would you have another look?

PatrickDekker98 commented on code/headers/rhino.hpp:
```diff
@@ -0,0 +1,56 @@
+         * @brief
+         * This function will set the speed between -100 and 100 % speed
+         * @detail
+         * the speed set must not exceed -127 of 127, if it is, it will set back
```
this is a nitpick, not something i would not request changes for, but 'of' should probably be 'or'

crisvdn replied:
Where do you actually read this? I read ```         * the speed set must not exceed -127 of 127, if it is, it will set back
         * to 127 or -127```

PatrickDekker98 replied:
my bad

PatrickDekker98 commented on code/main.cpp:
```diff
@@ -1,35 +1,41 @@
-    auto usart = r2d2::usart::hardware_usart_c<r2d2::usart::usart0>(9600);
+
+    // Create usart object.
+        auto usart = r2d2::usart::hardware_usart_c<r2d2::usart::usart0>(115200);
```
This line's indent is incorrect

crisvdn replied:
https://github.com/R2D2-2019/moving_platform/pull/30/commits/828469270908ab8c8bdaf7bc050cf363f1299915

PatrickDekker98 commented on code/src/beetle.cpp:
```diff
@@ -31,81 +31,163 @@ namespace r2d2::moving_platform {
         qik_2s12v10_motorcontroller.set_m0_speed(new_speed);
-        qik_2s12v10_motorcontroller.set_m1_speed(-new_speed);
+        qik_2s12v10_motorcontroller.set_m1_speed(new_speed);
     } // namespace r2d2::moving_platform
```
this comment suggests namespace r2d2::moving_platform ends here but it does not

crisvdn replied:
https://github.com/R2D2-2019/moving_platform/pull/30/commits/8070a9986ffcbc2c2bc39b0dd23e8bc5519e25c1

PatrickDekker98 commented on code/src/beetle.cpp:
```diff
@@ -31,81 +31,163 @@ namespace r2d2::moving_platform {
     }
-} // namespace r2d2::moving_platform
\ No newline at end of file
+    void beetle_c::move(uint16_t distance) {
```
there is a missing newline between two function definitions

crisvdn replied:
https://github.com/R2D2-2019/moving_platform/pull/30/commits/8070a9986ffcbc2c2bc39b0dd23e8bc5519e25c1

This is fixed. I don't understand why it doesn't say outdated though.

PatrickDekker98 commented on code/src/rhino.cpp:
```diff
@@ -0,0 +1,97 @@
+          motor_encoder_m0(encode_m0),
+          motor_encoder_m1(encode_m1) {
+    }
+    void rhino_c::set_speed(int8_t new_speed) {
```
missing newline at the end of a function declaration

crisvdn replied:
https://github.com/R2D2-2019/moving_platform/pull/30/commits/828469270908ab8c8bdaf7bc050cf363f1299915

This is fixed, I dont understand why it doesn't say outdated though.

PatrickDekker98 reviewed (changes requested):
sorry but there are too many nitpicks, most of which can be solved by running clang format once.

crisvdn committed:
Changed line indent location. Removed note

Change requests made: Indentation was wrong and ending namespace note, which shouldn't be there.

crisvdn committed:
Added clang format. fixed line indentation

PR request about line indentation and running clang

crisvdn committed:
Merge pull request #30 from R2D2-2019/patch-PR_26

Patch pr 26

crisvdn commented:
@PatrickDekker98 can you take another look? 
I've run clang as well on every file possible. Changes were made in patch-PR_26 that was merged.
